src/utils/HashUtils.ts

```diff
@@ -241,4 +232,24 @@ export class HashUtils {
       .compact()
       .value();
   }
+
+  private static removeSquareBrackets(value: string) {
+    if (HashUtils.startsWithEncodedLeftSquareBracket(value)) {
+      value = value.replace(Utils.safeEncodeURIComponent(HashUtils.DELIMITER.arrayStart), '');
```

Comment:
I agree generally, but I think it's okay in this case. Here the variable is a string which is a value not a pointer to an object or array where mutating the arg would have unexpected effects outside the scope of the method.

I could add a line at the top of the method:
```
let sanitizedValue = value;
if (condition) {
    sanitizedValue = sanitizedValue.replace(regex, '');
}
etc.
...
return sanitizedValue
```

but I don't think it would make a difference. What do you think?